colossalai/pipeline/policy/bert.py

```diff
@@ -221,34 +241,33 @@ def custom_forward(*inputs):
         pooled_output = self.pooler(sequence_output) if self.pooler is not None else None
         if not return_dict:
             return (sequence_output, pooled_output) + layer_outputs[1:]
+        # return dict is not supported at this moment
+        else:
+            return BaseModelOutputWithPastAndCrossAttentions(
+                last_hidden_state=hidden_states,
+                past_key_values=next_decoder_cache,
+                hidden_states=all_hidden_states,
+                attentions=all_self_attentions,
+                cross_attentions=all_cross_attentions,
+            )
 
-    # output of non-first and non-last stages:
-    if not return_dict:
-        return tuple(v for v in [
-            hidden_states,
-            next_decoder_cache,
-            all_hidden_states,
-            all_self_attentions,
-            all_cross_attentions,
-        ] if v is not None)
-
-    # return dict is not supported at this moment
-    return BaseModelOutputWithPastAndCrossAttentions(
-        last_hidden_state=hidden_states,
-        past_key_values=next_decoder_cache,
-        hidden_states=all_hidden_states,
-        attentions=all_self_attentions,
-        cross_attentions=all_cross_attentions,
-    )
+    # output of non-first and non-last stages: must be a dict
+    else:
+        # intermediate stage always return dict
+        return BertModelIntermediateOutput(hidden_states=hidden_states,)
```

Comment:
`BertModelIntermediateOutput` has compatibility problem with pytree, using original `dict` is better.

Reply:
done